mipengine/node/monetdb_interface/views.py

```diff
@@ -51,21 +51,29 @@ def create_view(
 
     db_execute(view_creation_query)
 
-    if check_min_rows:
-        view_rows_query_result = db_execute_and_fetchall(
-            f"""
-            SELECT COUNT(*)
-            FROM {view_name}
-            """
+    view_rows_query_result = db_execute_and_fetchall(
+        f"""
+        SELECT COUNT(*)
+        FROM {view_name}
+        """
+    )
+    view_rows_result_row = view_rows_query_result[0]
+    view_rows_count = view_rows_result_row[0]
+
+    if view_rows_count < 1:
+        db_execute(f"""DROP VIEW {view_name}""")
```

Comment:
If the node is already added in the cleaner before being removed in the `create_primary_data_views`, is there a need to drop the view here and in the elif?

Reply:
It is sure that the view has to be dropped, so I do not see why it should rely on an external procedure to do so